src/core/qgscopyfiletask.cpp

```diff
@@ -60,10 +60,10 @@ bool QgsCopyFileTask::run()
   const int chunkSize = std::clamp( size / 100, 1024, 1024 * 1024 );
 
   int bytesRead = 0;
-  char data[chunkSize];
+  std::unique_ptr<char> data( new char[chunkSize] );
```

Comment:
Would a `std::vector` be better qualified?

Reply:
yes! Done